File: 125. Valid Palindrome/125. Valid Palindrome.py
#
# Problem: 125. Valid Palindrome
# Difficulty: Easy
# Link: https://leetcode.com/problems/valid-palindrome/submissions/1983655031/
# Language: python
# Date: 2026-04-20


class Solution(object):
    def isPalindrome(self, s):
        str = '';
        length = len(s)
        for i in range(0,length):
            if s[i].isalnum():
                str += s[i].lower()

        str2 = str[::-1]
        return str2==str               
        

File: 125. Valid Palindrome/125. Valid Palindrome.cpp
/*
 * Problem: 125. Valid Palindrome
 * Difficulty: Easy
 * Link: https://leetcode.com/problems/valid-palindrome/submissions/1983644656/
 * Language: cpp
 * Date: 2026-04-20
 */

class Solution {
public:
    bool isPalindrome(string s) {
        string rev ;
        for(int i = 0;i<s.length();i++){
            if(isalnum(s[i])){
                rev += tolower(s[i]);
            }
        }
        string rev2 = rev;
        reverse(rev.begin(),rev.end());
        return rev==rev2;

    }
};
